lib/utils.js

```diff
@@ -336,18 +336,25 @@ function sanitizeQuery(query, options) {
   const prohibitedKeys = options.prohibitedKeys;
   const offendingKeys = [];
   const normalizeUndefinedInQuery = options.normalizeUndefinedInQuery;
-  const maxDepth = options.maxDepth || 10;
+  const maxDepth = options.maxDepth || 12;
   // WARNING: [rfeng] Use map() will cause mongodb to produce invalid BSON
   // as traverse doesn't transform the ObjectId correctly
   const result = traverse(query).forEach(function(x) {
     /**
      * Security risk if the client passes in a very deep where object
      */
+    if (this.circular) {
+      const msg = g.f('The query object is circular');
+      const err = new Error(msg);
+      err.statusCode = 400;
+      err.code = 'QUERY_OBJECT_IS_CIRCULAR';
+      throw err;
+    }
     if (this.level > maxDepth || this.circular) {
```

Comment:
Can we remove the check for `this.circular` in this if condition here since we've separated the checks?

Reply:
fixed